domain/schema/model/sql/0017-machine.sql

```diff
@@ -154,3 +154,14 @@ CREATE TABLE machine_removals (
     FOREIGN KEY (machine_uuid)
     REFERENCES machine (uuid)
 );
+
+-- machine_lxd_profile table keeps track of the lxd profiles (previously charm-profiles)
+-- for a machine.
+CREATE TABLE machine_lxd_profile (
+    machine_uuid TEXT NOT NULL,
+    name TEXT NOT NULL,
+    PRIMARY KEY (machine_uuid, name),
```

Comment:
There is a potential pitfall here. The order in which lxd profiles are applied is important, as each successive profile over writes items defined prior profiles. The order should not change.

When a new profile is applied, it's to the end of the list. This is problematic with upgrading charms as the name of the profile includes the charm revision.

A caveat is that we always start the list of profiles for a machine with the `default` profile and `juju-<model>` profile. This never changes.

Reply:
You could cheat here and when you get the `machine_lxd_profile` you sort by the magic rowid column.

But in reality, you need to add an index and just use the slice index when adding, and ordering it when it comes out.

Reply:
Nice! yeah we were just discussing this with @hmlanigan and I was even thinking about adding a new column just for sorting. Because the worst case scenario was to switch back to having one row per machine and profile names would be an array somehow.

Reply:
There is precedence of this:

https://github.com/juju/juju/blob/3795cb2a3e8745b3f88308cc0b5f02215d96bf1b/domain/schema/model/sql/0015-charm.sql#L511

Reply:
Done, see 82a3a302d8 @hmlanigan and @SimonRichardson.